spec/request_handler/body_parser_spec.rb

```diff
@@ -244,6 +255,138 @@
     it_behaves_like 'flattens the body as expected'
   end
 
+  context 'with included_schemas defined' do
+    let(:included_schemas) do
+      {
+        people:   schema,
+        comments: schema
+      }
+    end
+    let(:raw_body) do
+      <<-JSON
+      {
+        "data": {
+          "type": "articles",
+          "id": "1",
+          "attributes": {
+            "title": "JSON API paints my bikeshed!"
+          },
+          "relationships": {
+            "author": {
+              "data": {
+                "type": "people",
+                "id": "9"
+              }
+            },
+            "comments": {
+              "data": [
+                {
+                  "type": "comments",
+                  "id": "5"
+                },
+                {
+                  "type": "comments",
+                  "id": "12"
+                }
+              ]
+            }
+          }
+        },
+        "included": [
+          {
+            "type": "people",
+            "id": "9",
+            "attributes": {
+              "first-name": "Dan",
+              "last-name": "Gebhardt",
+              "twitter": "dgeb"
+            }
+          },
+          {
+            "type": "comments",
+            "id": "5",
+            "attributes": {
+              "body": "First!"
+            },
+            "relationships": {
+              "author": {
+                "data": {
+                  "type": "people",
+                   "id": "2"
+                }
+              }
+            }
+          },
+         {
+            "type": "comments",
+            "id": "12",
+            "attributes": {
+              "body": "I like XML better"
+            },
+            "relationships": {
+              "author": {
+                "data": {
+                  "type": "people",
+                  "id": "9"
+                }
+              }
+            }
+          }
+        ]
+      }
+      JSON
+    end
+    let(:wanted_result) do
+      [
+        {
+          'type' => 'articles',
+          'id' => '1',
+          'title' => 'JSON API paints my bikeshed!',
+          'author' => {
+            'type' => 'people',
+            'id' => '9'
+          },
+          'comments' => [
+            {
+              'type' => 'comments',
+              'id' => '5'
+            },
+            {
+              'type' => 'comments',
+              'id' => '12'
+            }
+          ]
+        },
+        {
+          'type' => 'people',
+          'id' => '9',
+          'first-name' => 'Dan',
```

Comment:
please change the example to snake_case attributes. We probably need to add some extra handling for dashed attributes. #14